drivers/volume/portworx/portworx.go

```diff
@@ -1039,27 +1039,45 @@ func (d *portworx) StartDriver(n node.Node) error {
 }
 
 func (d *portworx) UpgradeDriver(version string) error {
+	customImageRegex := regexp.MustCompile(`oci=(?P<oci>([\w][\w._-]+\/)?[\w][\w._-]*\:[\w][\w.-]*),px=(?P<px>([\w][\w._-]+\/)?[\w][\w._-]*\:[\w][\w.-]*)`)
```

Comment:
can u give examples of what this regex matches to?

Reply:
any valid docker image in the following 
oci=<any valid docker image>,px=<any valid docker image>

i.e
oci=repo/image:tag,px=repo/image:tag
oci=repo/image,px=repo/image
oci=image:tag,px=image:tag
oci=image,px=image

Reply:
Make version a proper input type (struct) instead of a special string pattern that the implementation must parse.

The struct can have the oci and px image as items.

For any interface, the function definition must be clear and the arguments must be self-documenting so that users can know how to invoke them.

Reply:
correct me if I'm wrong, but volume interface is driver agnostic, so if we add a different structure we might tie it to portworx driver

Reply:
Add a generic input which has list of images. Each item has an image type and image version. Both of these are strings.